# Assignment-1: Revenue Estimation

In [1]:
import numpy as np
import pandas as pd
from scipy import stats

In [2]:
df=pd.read_csv("E:\data\data_set\Revenue_forcasting_CI.csv")

In [3]:
df

,Age,Marketing_Exposure,Previous_Purchases,Monthly_Revenue
0,58,1,1,2494.941406
1,48,0,9,2356.387522
2,34,1,11,1978.357376
3,27,1,16,3634.738976
4,40,0,7,2268.067368
...,...,...,...,...
195,41,1,5,2507.118415
196,30,1,6,2145.153525
197,35,1,19,3569.311825
198,52,0,8,2507.930324


# population parameters (mean,std)

In [4]:
pop_mean=df['Monthly_Revenue'].mean()
pop_mean

np.float64(2664.6927557651634)

In [5]:
# variance population
deviation=df['Monthly_Revenue']-pop_mean

In [6]:
deviation

0      -169.751350
1      -308.305234
2      -686.335379
3       970.046220
4      -396.625387
          ...     
195    -157.574340
196    -519.539231
197     904.619069
198    -156.762432
199    1289.805076
Name: Monthly_Revenue, Length: 200, dtype: float64

In [7]:
deviationsquare=deviation**2
deviationsquare=sum(deviationsquare)

In [8]:
variance=deviationsquare/df.shape[0]

In [9]:
variance

370205.51561214664

# standard deviation of population

In [10]:
std=np.sqrt(variance)
std

np.float64(608.4451623705678)

# sample of statestics(mean,std)

In [11]:
np.random.seed(42)
sample=df.sample(60)

In [12]:
sample

,Age,Marketing_Exposure,Previous_Purchases,Monthly_Revenue
95,28,1,18,3836.996695
15,41,1,16,3484.169727
30,22,0,8,2821.787011
158,46,1,9,2364.456730
128,49,1,4,1859.854436
115,45,1,18,4216.071263
69,40,1,15,3453.777816
170,22,0,13,3431.406783
174,21,1,5,1546.215989
45,26,0,19,2281.077588


In [13]:
sample_mean=sample['Monthly_Revenue'].mean()
sample_mean

np.float64(2724.451737112798)

# sample standard deviation

In [14]:
sample_devaition=sample['Monthly_Revenue']-sample_mean
sample_devaition

95     1112.544958
15      759.717990
30       97.335274
158    -359.995007
128    -864.597301
115    1491.619526
69      729.326079
170     706.955046
174   -1178.235748
45     -443.374149
66      644.978256
182    -232.996941
165    -871.081795
78     -506.477366
186      12.531427
177     -19.140711
56     -479.323289
152     558.972795
82      113.279887
68      -91.236969
124    -186.427717
16      552.894660
148     521.593453
93      150.334628
65      283.360417
60     -382.344659
84      244.084523
67       88.165939
125     652.028251
132   -1047.380117
9      -660.440164
18      -26.562631
55      594.127495
75      -13.991852
150    -200.911389
104     109.862505
135   -1154.614250
137    -674.509566
164    -825.727296
76     -510.503620
79       72.752907
197     844.860088
38      171.459286
24      754.956580
122    -252.600403
195    -217.333322
29     -726.372296
19      406.334184
143      37.932366
86     -167.595337
114    -861.892988
173     671.000011
5       705.

In [15]:
squra_sampl_deviation=sum(sample_devaition**2)

In [16]:
sample_var=squra_sampl_deviation/sample.shape[0]
sample_var

382375.8026552312

In [17]:
sample_std=np.sqrt(sample_var)
sample_std

np.float64(618.3654280886273)

In [18]:
## varian by function

In [19]:
df['Monthly_Revenue'].var()

372065.8448363282

In [20]:
# variance by calculation
variance

370205.51561214664

In [21]:
# std by function
df['Monthly_Revenue'].std()

609.9720033217329

In [22]:
# std by calculation
std

np.float64(608.4451623705678)

In [23]:
# all results 
print(f'pupulation mean is: {pop_mean}, and standard devaiton is: {std}')
print(f'sample mean is: {sample_mean}, and standard devaiton is: {sample_std}')

pupulation mean is: 2664.6927557651634, and standard devaiton is: 608.4451623705678
sample mean is: 2724.451737112798, and standard devaiton is: 618.3654280886273


# T-Test for sample data

### Q3: Construct 95% Confidence Interval (Manual Formula)

In [24]:
sample_mean
degreefre=sample.shape[0]-1
alpha=0.05



In [25]:
stderror=sample_std/np.sqrt(60)
stderror

np.float64(79.83063349525578)

In [26]:
# I have to find t-value corrosponding to degree of fredom
t_value=1-0.05/2
t_value # find the t-value of 0.975 corrospond to 59

0.975

In [27]:
t_value=2.00

In [28]:
# the positive interval
posi=sample_mean+t_value*stderror
posi

np.float64(2884.11300410331)

In [29]:
# negetive interval
negi=sample_mean-t_value*stderror
negi

np.float64(2564.7904701222865)

# margin of error

In [30]:

marginoferror=t_value*stderror
marginoferror

np.float64(159.66126699051156)

# confidence interval

In [31]:
print(f' The confidence interval for 95% and 0.05 aplha is min :{negi} - mean {sample_mean} - max {posi}')

 The confidence interval for 95% and 0.05 aplha is min :2564.7904701222865 - mean 2724.451737112798 - max 2884.11300410331


In [32]:
# its nothing but
sample_mean-marginoferror,sample_mean+marginoferror

(np.float64(2564.7904701222865), np.float64(2884.11300410331))

### Built-in function to calculate confidence interval

from scipy import stats
interval = stats.t.interval(confidence, df, loc=xbar, scale=SEM)
confidence: Probability (0.95 for 95% CI).

df: n-1.

loc: Sample mean (x̄).

scale: Standard error (s/sqrt(n)

In [33]:
interval=stats.t.interval(0.95,59,sample_mean,stderror)
interval

(np.float64(2564.711008458933), np.float64(2884.192465766663))

# Business Interpretation

- The average montly revenue per customer in city is lies b/w **2564.71 to 2884.19** with **95%** confidence
- The the **Point Estimate** is 2724.45
- The **Margin of Error** is 159.66

# Z-test for Population
- In real scenario we doesn't perform z-test
- because we doesn't have the population standard deviation  

In [34]:
pop_mean #muuu

np.float64(2664.6927557651634)

In [35]:
std # sigma

np.float64(608.4451623705678)

In [36]:
standard_error=std/np.sqrt(df.shape[0])
standard_error

np.float64(43.023570029237845)

# 95% confidence
- aplha 0.05
- standard normal variable(z)=xi-xbar/std

In [37]:
z_value=sum(df['Monthly_Revenue']-pop_mean)/std
z_value

np.float64(1.8684812494634664e-14)

### now i have to search z value corrosponding to 1.86 and  at 95% confidence

In [38]:
z_score=0.9686

In [39]:
maximum=pop_mean+z_score*standard_error
maximum

np.float64(2706.3653856954834)

In [40]:
minimum=pop_mean-z_score*standard_error
minimum

np.float64(2623.0201258348434)

# Assigment-2: Churn Rate Estimation

In [41]:
df=pd.read_csv("E:\data\data_set\Churn_Rate_Estimation.csv")

In [42]:
df

,Tenure_Months,Monthly_Usage_Hours,Support_Tickets,Churn
0,20,53.956772,2,0
1,24,47.169046,1,0
2,13,33.248742,1,0
3,41,34.377245,4,0
4,42,55.840686,3,1
...,...,...,...,...
395,4,48.233259,1,0
396,25,41.362063,3,1
397,13,36.893949,1,1
398,5,42.524743,5,1


### Point Estimate of churn rate 
- point estimate means mean of the churn rate

In [43]:
point_est=df['Churn'].mean()
point_est

np.float64(0.5325)

## Q7: Construct 95% Confidence Interval for Proportion

# Sample Propertion
p=x/n , x - number of succefull outcomes ,n-total number of outcomes

In [44]:
churn_1=df[df['Churn']==1]
churn_1

,Tenure_Months,Monthly_Usage_Hours,Support_Tickets,Churn
4,42,55.840686,3,1
5,8,47.741120,1,1
8,2,44.416944,0,1
10,29,51.815661,2,1
11,15,42.729358,2,1
...,...,...,...,...
391,30,38.585733,3,1
396,25,41.362063,3,1
397,13,36.893949,1,1
398,5,42.524743,5,1


In [45]:
sample_proportion=churn_1.shape[0]/df.shape[0]
sample_proportion

0.5325

In [46]:
standard_error=np.sqrt(sample_proportion*(1-sample_proportion)/df.shape[0])
standard_error

np.float64(0.024947131598642758)

# Z-value
- standardization of values

In [47]:
z_value=sum((df['Churn']-point_est)/df['Churn'].shape[0])
z_value

2.3852447794681098e-17

In [48]:
z_score=0.9911

In [49]:
margin_of_error=z_score*standard_error
margin_of_error

np.float64(0.024725102127414838)

# Confidence Interval

In [50]:
mini=point_est-margin_of_error
mini

np.float64(0.5077748978725851)

In [51]:
maxi=point_est+margin_of_error
maxi

np.float64(0.5572251021274148)

In [52]:
point_est

np.float64(0.5325)

In [53]:
# confidence interval
print(f'confidence interval is lower bound {mini} mean {point_est} upper bound{maxi}')

confidence interval is lower bound 0.5077748978725851 mean 0.5325 upper bound0.5572251021274148


- Q8: Interpret Business Impact
- If company has 50,000 users:
    - Estimate number of churned users
    - Estimate possible range of churned users using CI

# Estimate number of churned users

In [68]:
# minimum
print(f'{500000*mini}')

253887.44893629255


In [67]:
# maximum
print(f'{500000*maxi}')

278612.5510637074


In [65]:
print(f'Estimate churned {500000*point_est}')

Estimate churned 266250.0


In [61]:
df[df['Churn']==1].shape[0]/df.shape[0]

0.5325

- At the 95 percent of confidence from the sample
- the estimated number of churned users is 266250
- 50.77 percentage of minimum customers estimated to churn
- 55.72 precentage of maximum customers estimated to churn